code/modules/mod/modules/modules_general.dm

```diff
@@ -223,11 +223,19 @@
 	var/min_range = 2
 	/// Maximum range we can set.
 	var/max_range = 5
+	/// The cooldown before we can re-activate this after having it forcefully extinguished
+	COOLDOWN_DECLARE(activation_cooldown)
 
 /obj/item/mod/module/flashlight/on_activation()
+	if(!COOLDOWN_FINISHED(src, activation_cooldown))
+		to_chat(mod.wearer, "<span class='warning'>[src] isn't ready after being shut down!</span>")
+		return
 	. = ..()
 	if(!.)
 		return
+
+	COOLDOWN_RESET(src, activation_cooldown)
```

Comment:
This cooldown isnt technically necessary, but it will make it clearer to anyone viewing the vars, so I think its probably fine